### PR TITLE
fix(lsp): type-errors and other nits in vim.lsp.log

### PR DESCRIPTION
**Pull request by MariaSolOs**
Small nits that I don't want to include in other PRs as unrelated changes.

### CODE REVIEW

File: runtime/lua/vim/lsp/log.lua [hunk 1]
```diff
@@ -32,12 +32,12 @@ local function notify(msg, level)
   end
 end
 
-local logfilename = vim.fs.joinpath(vim.fn.stdpath('log'), 'lsp.log')
+local logfilename = vim.fs.joinpath(vim.fn.stdpath('log') --[[@as string]], 'lsp.log')
```
Comment (justinmk) [inline runtime/lua/vim/lsp/log.lua:35]: does LuaLs support parameterizing the return type based on the input?

alternatively, maybe we should just fib a little and define `stdpath():string` in the meta files. Almost no one ever uses the `_dirs` cases.

**Review by justinmk — APPROVED** [review]

File: runtime/lua/vim/lsp/log.lua (hunk 1 above)
Comment (MariaSolOs) [reply runtime/lua/vim/lsp/log.lua:35]: I believe Lua's type annotations aren't powerful enough for that yet. Maybe we could use overloads but the type for this one is autogenerated and idk if we have support for that yet (but @lewis6991 would know better).

Comment (lewis6991) [reply runtime/lua/vim/lsp/log.lua:35]: Return type polymorphism is generally difficult for a type system, so most don't allow it. The only exception is generics.

Comment (justinmk) [reply runtime/lua/vim/lsp/log.lua:35]: Let's define stdpath as returning `string` then

**Pull request merged**